packages/remix-gcloud-functions-gen2/server.ts

```diff
@@ -0,0 +1,164 @@
+import type { NextFunction } from "express";
+import type { AppLoadContext, ServerBuild } from "@remix-run/server-runtime";
+import type {
+  Request as GcfRequest,
+  Response as GcfResponse,
+} from "@google-cloud/functions-framework";
+import type { 
+  RequestInit as NodeRequestInit,
+  Response as NodeResponse
+ } from "@remix-run/node";
+import {
+  createRequestHandler as createRemixRequestHandler,
+  Headers as NodeHeaders,
+  Request as NodeRequest,
+  writeReadableStreamToWritable
+} from "@remix-run/node";
+import {
+  // This has been added as a global in node 15+
+  AbortController,
+} from "@remix-run/node";
+import { Readable } from 'stream'
+
+/**
+ * A function that returns the value to use as `context` in route `loader` and
+ * `action` functions.
+ *
+ * You can think of this as an escape hatch that allows you to pass
+ * environment/platform-specific values through to your loader/action.
+ */
+export interface GetLoadContextFunction {
+  (req: GcfRequest, res: GcfResponse): AppLoadContext;
+}
+
+//export type RequestHandler = ReturnType<typeof createRequestHandler>;
+export type RequestHandler = (
+  req: GcfRequest,
+  res: GcfResponse,
+  next: NextFunction
+) => Promise<void>;
+
+/**
+ * Returns a request handler for Express that serves the response using Remix.
+ */
+export function createRequestHandler({
+  build,
+  getLoadContext,
+  mode = process.env.NODE_ENV,
+}: {
+  build: ServerBuild;
+  getLoadContext?: GetLoadContextFunction;
+  mode?: string;
+}): RequestHandler {
+  let handleRequest = createRemixRequestHandler(build, mode);
+
+  return async (req: GcfRequest, res: GcfResponse, next: NextFunction) => {
+    try {
+      let request = createRemixRequest(req);
+      let loadContext =
+        typeof getLoadContext === "function"
+          ? getLoadContext(req, res)
+          : undefined;
+
+      let response = (await handleRequest(
+        request,
+        loadContext
+      )) as NodeResponse;
+
+      await sendRemixResponse(res, response);
+    } catch (error) {
+      // Express doesn't support async functions, so we have to pass along the
+      // error manually using next().
+      next(error);
```

Comment:
Pretty sure this will fail as Google Cloud functions don't provide a next function.

https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/759aac18ad297363c4651efcf19ab2bf8fa9625a/src/functions.ts#L46-L48

I was working on an adapter independently, and throw an error at this point in my version. Likely needs a test though https://github.com/remix-run/remix/pull/3397